### PR TITLE
Mentor/tests

### PR DESCRIPTION
**Pull request by NathanGrenier**
### 🛠 Proposed Changes:

Added testing infrastructure.

### 📝 Details:

- Added sample test
- Added docs on generating coverage
- Added tests and coverage report to workflow

### CODE REVIEW

File: .github/workflows/integrate.yml [hunk 1]
```diff
@@ -30,6 +30,34 @@ jobs:
...
+          python3 -m robotpy test
+      - name: Generate coverage report
+        run: |
+          python3 -m robotpy coverage test
+          coverage html -d coverage_html
+      - name: HTML Preview
+        id: html_preview
+        uses: pavi2410/html-preview-action@v2
+        with:
+          html_file: "index.html"
```
Comment (NathanGrenier) [inline .github/workflows/integrate.yml:60]: This [action](https://github.com/marketplace/actions/html-preview) currently isn't working.

There may be another way to host the generated coverage html file on github.

Comment (NathanGrenier) [inline .github/workflows/integrate.yml:60]: This [action](https://github.com/marketplace/actions/html-preview) currently isn't working.

There may be another way to host the generated coverage html file on github.

Comment (Portnord) [reply .github/workflows/integrate.yml:60]: Can we convert this pull request to a draft?

Comment (NathanGrenier) [reply .github/workflows/integrate.yml:60]: not sure what a draft is.

This PR can go through as long as I remove the `code report` and `html preview` portion of the `integrate.yml` workflow.

**Pull request merged**